hobbits/src/main/kotlin/org/apache/tuweni/hobbits/Message.kt

```diff
@@ -38,48 +40,29 @@ class Message(
      * Reads a message from a byte buffer.
      * @param message the message bytes
      * @return the message interpreted by the codec, or null if the message is too short.
+     * @throws IllegalArgumentException if the message doesn't start with the correct preamble.
      */
     @JvmStatic
     fun readMessage(message: Bytes): Message? {
-      var requestLineBytes: Bytes? = null
-      for (i in 0 until message.size()) {
-        if (message.get(i) == '\n'.toByte()) {
-          requestLineBytes = message.slice(0, i)
-          break
-        }
-      }
-      if (requestLineBytes == null) {
+      if (message.size() < MESSAGE_HEADER_LENGTH) {
         return null
       }
-      val requestLine = String(requestLineBytes.toArrayUnsafe(), StandardCharsets.UTF_8)
-      val segments = Splitter.on(" ").split(requestLine).iterator()
-
-      val protocol = segments.next()
-      if (!segments.hasNext()) {
-        return null
-      }
-      val version = segments.next()
-      if (!segments.hasNext()) {
-        return null
-      }
-      val command = segments.next()
-      if (!segments.hasNext()) {
-        return null
-      }
-      val headersLength = segments.next().toInt()
-      if (!segments.hasNext()) {
-        return null
+      if (message.slice(0, PREAMBLE.size) != Bytes.wrap(PREAMBLE)) {
+        throw IllegalArgumentException("Message doesn't start with correct preamble")
       }
-      val bodyLength = segments.next().toInt()
+      val version = message.getInt(PREAMBLE.size)
+      val protocol = Protocol.fromByte(message.get(PREAMBLE.size + 4))
+      val headersLength = message.getInt(PREAMBLE.size + 4 + 1)
+      val bodyLength = message.getInt(PREAMBLE.size + 4 + 1 + 4)
 
-      if (message.size() < requestLineBytes.size() + 1 + headersLength + bodyLength) {
+      if (message.size() < PREAMBLE.size + java.lang.Integer.BYTES * 3 + 1 + headersLength + bodyLength) {
```

Comment:
just use: `MESSAGE_HEADER_LENGTH` instead of `PREAMBLE.size + java.lang.Integer.BYTES * 3 + 1`